File: array/js/second-largest-element.js
const getSecondLargestElement = (arr) => {
  let firstLargest = -1, secondLargest = -1;

  for (const e in arr) {
    if (arr[e] > firstLargest) {
      secondLargest = firstLargest;
      firstLargest = arr[e];
    } else if (arr[e] > secondLargest && arr[e] < firstLargest) {
      secondLargest = arr[e];
    }
  }

  console.log('Second largest:', secondLargest);
  return secondLargest;
};

const arr = [12, 35, 1, 10, 34, 1];
console.log(getSecondLargestElement(arr));

File: array/rust/src/second_largest_element.rs
pub fn get_second_largest(arr: &[i32]) -> i32 {
    let mut largest = -1;
    let mut second_largest = -1;

    for e in arr.iter() {
        if *e > largest {
            second_largest = largest;
            largest = *e;
        } else if *e > second_largest {
            second_largest = *e;
        }
    }

    println!("Seccond largest element: {}", second_largest);
    second_largest
}
